test/unit/mriController.test.js

```diff
@@ -0,0 +1,343 @@
+var assert = require("assert");
+const uploadController = require('../../controller/mri/upload.controller');
+const mriController = require('../../controller/mri/mri.controller');
+const monk = require('monk');
+require('mocha-sinon');
+const sinon = require('sinon');
+var db = monk('http://localhost:27017/brainbox');
+
+describe('MRI Controller: ', function () {
+    it('Validator function should perform the validations correctly', async function () {
+        let req = {
+            body: {
+                url: 'abc.com',
+                atlasName: 'MyAtlas',
+                atlasProject: 'Visualisation@',
+                atlasLabelSet: 'SampleLabelSet',
+                token: 'jnqpincpienfcpewnfcpewn123'
+            },
+            value: 0,
+            validationErrors: function() {
+                console.log('Testing!');
+            }
+        };
+        let validate = sinon.stub(req, 'validationErrors').resolves(true);
+        let resSpy = sinon.spy();
+
+        let res = {
+            status: sinon.stub().returns({ send: resSpy }),
+        };
+        await mriController.validator(req, res, function () {
+            console.log('Unit Test finished!');
+        });
+        assert.notDeepStrictEqual(validate.calledOnce, true);
+        assert.notDeepStrictEqual(resSpy.calledOnce, true);
+        assert.notDeepStrictEqual(resSpy.calledWith(null), true);
+        sinon.restore();
+    });
+
+    it('validatorPost function() should perform the validations correctly', async function () {
+        let reqSpy = sinon.spy();
+        let urlSpy = sinon.spy();
+        let req = {
+            body: {
+                url: 'abc.com',
+                atlasName: 'MyAtlas',
+                atlasProject: 'Visualisation@',
+                atlasLabelSet: 'SampleLabelSet',
+                token: 'jnqpincpienfcpewnfcpewn123'
+            },
+            value: 0,
+            validationErrors: function() {
+                return;
+            },
+            checkBody: sinon.stub().returns({ notEmpty: reqSpy, isURL: urlSpy })
+        };
+        let resSpy = sinon.spy();
+
+        let res = {
+            status: sinon.stub().returns({ send: sinon.stub().returns({ end: resSpy })}),
+        };
+        await mriController.validatorPost(req, res, () => {});
+        assert.strictEqual(reqSpy.callCount, 1);
+        assert.strictEqual(urlSpy.callCount, 1);
+        assert.strictEqual(resSpy.callCount, 0);
+        sinon.restore();
+    });
+
+    it('download MRI function should download the file when correct input is given', async function() {
```

Comment:
A call to the mri() function does not download the file, but the entry in the mongoDB associated with that file. Change the test name to "should get the MRI record associated with the URL"